### ISSUE TITLE
Handle unbounded fonts

### ISSUE THREAD

**Issue by CollinHeist** [CollinHeist/TitleCardMaker#43]
For cards that support custom fonts (notably the `StandardTitleCard` class), if the given font has [unbounded text](https://legacy.imagemagick.org/Usage/text/#overflow) - an example of which is the South Park font:

<img src="https://i.ibb.co/X4CxrpC/out.jpg" width="300">

Here, the text extends well beyond the bounding box (in red). This results in bad offsets during `-annotate` calls, and overall poorly formatted cards like:

<img src="https://i.ibb.co/YkgCCd7/South-Park-1997-S25-E06.jpg" width="600">

The Maker ideally should have some way of handling or identifying these situations

**Comment by CollinHeist** [comment]
For the given command and resulting image:
```console
$ convert -size 500x300 xc:black -font south-park.ttf -pointsize 100 -gravity center -fill white -undercolor red -annotate +0+0 "testing" out.jpg
```
![ref0](https://user-images.githubusercontent.com/17693271/159082205-72c3a2b0-eb70-40ca-9485-8b78d5428ce7.jpg)

The vertical overflow is very evident. Passing this command through metrics gives:

```console
$ convert -debug annotate -size 500x300 xc:black -font south-park.ttf -pointsize 100 -gravity center -annotate +0+0 "testing" null: 2>&1 | grep Metrics:                            
  Metrics: text: testing; width: 436; height: 65; ascent: 72; descent: 21; max advance: 100; bounds: 0,-0.21875  64.1094,80.4219; origin: 436,0; pixels per em: 100,100; underline position: -15.4785; underline thickness: 6.00586
```

Using these reported widths and heights, I marked up the given bounding boxes, blue being the height, green being the ascent:

```console
$ convert -size 500x300 xc:black -font south-park.ttf -pointsize 100 -gravity center -fill white -undercolor red -annotate +0+0 "testing" -stroke blue -strokewidth 1.5 -fill None -draw "rectangle 32,117.5,468,182.5" -stroke green -strokewidth 1.5 -draw "rectangle 32,114,468,186" out.jpg
```
![out](https://user-images.githubusercontent.com/17693271/159084292-ba48bfab-acf3-4bf3-a7d6-4295d823b755.jpg)

Clearly, neither reported height metric corresponds to the actual text are of this font😢

**Comment by CollinHeist** [comment]
Looked into using fonttools, see below code

```python3
font = TTFont('south-park.ttf')
cmap = font['cmap']
t = cmap.getBestCmap()
s = font.getGlyphSet()
units_per_em = font['head'].unitsPerEm

def get_height(char, font=font, point_size=100):
    conversion = float(point_size) / units_per_em
    if 'glyf' in font:
        glyf = font['glyf'][t[ord(char)]]
        return (glyf.yMax-glyf.yMin)*conversion

    return None
```

Which seems to return the appropriate height:
![out](https://user-images.githubusercontent.com/17693271/159151006-97e10edd-9010-45ce-b152-68578705e34a.jpg)
 
However, as seen above, unbounded fonts don't necessarily have centered offsets.. Meaning this would (at best) be a half fix, and would require a huge rework of parsing each font/character set.. Not great

----

Instead, I implemented a more *manual* approach in #44 allowing for specifying a vertical offset and line spacing as a part of the series font. This is much more tedious, but is much less overhead.
